driver-core/src/test/java/com/datastax/driver/core/policies/PerHostErrorTrackerTest.java

```diff
@@ -0,0 +1,256 @@
+/*
+ *      Copyright (C) 2012-2015 DataStax Inc.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+package com.datastax.driver.core.policies;
+
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.Host;
+import com.datastax.driver.core.HostDistance;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.ScassandraCluster;
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.Statement;
+import com.datastax.driver.core.exceptions.UnauthorizedException;
+import org.scassandra.Scassandra;
+import org.testng.annotations.Test;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.scassandra.http.client.PrimingRequest.queryBuilder;
+import static org.scassandra.http.client.Result.success;
+import static org.scassandra.http.client.Result.unauthorized;
+
+public class PerHostErrorTrackerTest {
```

Comment:
Hrmm yes, that does make it much less deterministic.   It looks like Meters update their rates every 5 seconds (`Meter.TICK_INTERVAL`), although you can construct `Meter` instances with your own `Clock` implementation, which would give you more control over the ticking (which gets calculated everytime you mark the Meter).

What do you think about adding a withClock method to the ErrorAwarePolicyBuilder?  This would allow us to have more fine grained control over the `Meter` tick interval, which will allow us to have more deterministic tests.